onnx/defs/map/defs.cc

```diff
@@ -0,0 +1,402 @@
+          const auto& input_shape = ctx.getInputType(1)->sequence_type().elem_type().tensor_type().shape();
+          *(output_tensor_type->mutable_shape()) = input_shape;
+          
+        }));
+
+static const char* MapKeys_ver18_doc = R"DOC(
+Outputs a tensor that consists of all the 'keys' present
+in the map structure.
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    MapKeys,
+    18,
+    OpSchema()
+        .SetDoc(MapKeys_ver18_doc)
+        .Input(0, "map", " Map.", "M")
+        .Output(0, "keys", "Keys.", "T")
+        .TypeConstraint(
+            "M",
+            all_map_types(),
+            "Constrain input to a map type.")
+        .TypeConstraint(
+            "T",
+            OpSchema::all_map_key_tensor_types(),
+            "Constrain output type to integral and string tensor types.")
+        .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+          const auto input_type = ctx.getInputType(0);
+          if (nullptr == input_type) {
+            fail_type_inference("Input map is expected to have type info. Current type is null.");
+          }
+          ctx.getOutputType(0)->mutable_tensor_type()->set_elem_type(input_type->map_type().key_type());
+
+          const size_t numOutputs = ctx.getNumOutputs();
+          if (numOutputs != 1) {
+            fail_type_inference("MapKeys is expected to have 1 output.");
+          }
+
+          if (!hasNInputShapes(ctx, static_cast<int>(ctx.getNumInputs()))) {
+            return;
+          }
+
+          const auto& input_shape = ctx.getInputType(0)->map_type().value_type().sequence_type().elem_type().tensor_type().shape();
+          auto final_output_shape = ctx.getOutputType(0)->mutable_tensor_type()->mutable_shape();
+          final_output_shape->add_dim()->set_dim_value(input_shape.dim_size());
+        }));
+
+
+static const char* MapValues_ver18_doc = R"DOC(
+Outputs a tensor that consists of all the 'values' present
+in the map structure.
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    MapValues,
+    18,
+    OpSchema()
+        .SetDoc(MapValues_ver18_doc)
+        .Input(0, "map", " Map.", "M")
+        .Output(0, "values", "Sequence enclosing the values.", "S")
+        .TypeConstraint(
+            "M",
+            all_map_types(),
+            "Constrain input to a map type.")
+        .TypeConstraint(
+            "S",
+            OpSchema::all_tensor_sequence_types(),
+            "Constrain output to a sequence of any tensor type.")
+        .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+          const auto input_type = ctx.getInputType(0);
+          if (nullptr == input_type) {
+            fail_type_inference("Input map is expected to have type info. Current type is null.");
+          }
+          ctx.getOutputType(0)->CopyFrom(input_type->map_type().value_type());
+
+          const size_t numOutputs = ctx.getNumOutputs();
```

Comment:
These checks are unnecessary, and will be done automatically (given the information in the schema).